rmw_fastrtps_shared_cpp/src/rmw_publisher.cpp

```diff
@@ -20,6 +20,9 @@
 #include "rmw/impl/cpp/macros.hpp"
 #include "rmw/rmw.h"
 
+#include "fastdds/dds/publisher/DataWriter.hpp"
+#include "fastdds/dds/publisher/Publisher.hpp"
```

Comment:
```suggestion
#include "fastdds/dds/publisher/qos/DataWriterQos.hpp"
```

Header `Publisher.hpp` is not being used. Instead `DataWriterQos` is being used and it would be wise to include its header file.